### PR TITLE
feat(VFeatureDiscovery): add new component

### PR DESCRIPTION
**Pull request by dsseng**
## Description
A new 'Feature Discovery' component for Vuetify. :warning: Copied from #4952. Made by @JamieCurnow

## Motivation and Context
See #1171

## How Has This Been Tested?
In playground
TODO: unit

## Markup:

<details>

```vue
<template>
  <v-app>
    <v-content>
      <v-container >

        <v-layout row wrap justify-center>
          <v-flex xs6 text-xs-center>
            <v-card>
              <v-card-text>
                <v-layout row wrap justify-center>
                  <v-flex xs12 text-xs-center>
                    <h2>V Feature Discovery</h2>
                    <p>Currently overlay is not working since reqriting in TypeScript. Also should probably have a min height.</p>
                  </v-flex>
                </v-layout>

                <v-layout row wrap justify-center>
                  <v-spacer></v-spacer>
                  <v-switch label="top" v-model="top" style="flex-grow: 0"></v-switch>
                  <v-spacer></v-spacer>
                  <v-switch label="left" v-model="left" style="flex-grow: 0"></v-switch>
                  <v-spacer></v-spacer>
                  <v-switch label="flat" v-model="flat" style="flex-grow: 0"></v-switch>
                  <v-spacer></v-spacer>
                </v-layout>
                <v-layout row wrap justify-center>
                  <v-spacer></v-spacer>
                  <v-switch label="hide-overlay" v-model="hideOverlay" style="flex-grow: 0"></v-switch>
                  <v-spacer></v-spacer>
                  <v-switch label="persistent" v-model="persistent" style="flex-grow: 0"></v-switch>
                  <v-spacer></v-spacer>
                </v-layout>
                <v-layout row wrap justify-center>
                  <v-flex xs6 text-xs-center>
                    <v-text-field
                      type="number"
                      label="height"
                      v-model="height"
                    ></v-text-field>
                  </v-flex>
                </v-layout>
              </v-card-text>
              <v-card-actions>
                <v-spacer></v-spacer>
                <v-btn color="primary" @click="discover = !discover">Discover</v-btn>
                <v-spacer></v-spacer>
              </v-card-actions>
            </v-card>
          </v-flex>
        </v-layout>

        <v-feature-discovery
          class="white--text"
          color="primary"
          :top="top"
          :left="left"
          :hide-overlay="hideOverlay"
          :flat="flat"
          :persistent="persistent"
          :height="height"
          v-model="discover"
        >
          <h2 slot="title">Hey! New feature...</h2>
          <p slot="text">There's all sorts of new things that you can do with the great new features. Just click the buttons to check them out!</p>
          <div slot="actions">
            <v-btn color="white" depressed>Action</v-btn>
            <v-btn outline color="white" @click="discover = false">Lesser action</v-btn>
          </div>
        </v-feature-discovery>

      </v-container>
    </v-content>
  </v-app>
</template>

<script>
  export default {
    data: () => ({
      discover: false,
      top: false,
      left: false,
      hideOverlay: false,
      flat: false,
      persistent: false,
      height: 400
    })
  }
</script>

```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

**Review by dsseng — COMMENTED** [review]
Note: transition of highlight was broken by bb5ce4c

File: packages/vuetify/src/components/VFeatureDiscovery/VFeatureDiscovery.ts [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+    window.addEventListener('resize', this.updateRect)
+    window.addEventListener('keyup', this.keyPress)
+  },
+
+  methods: {
+    keyPress (e: KeyboardEvent) {
+      if (this.closeConditional() && e.keyCode === keyCodes.esc) this.isActive = false
+    },
+    updateTarget () {
+      if (this.targetEl) (this.targetEl as any).style.zIndex = ''
```
Comment (KaelWD) [inline packages/vuetify/src/components/VFeatureDiscovery/VFeatureDiscovery.ts]: Why as any?

File: packages/vuetify/src/components/VFeatureDiscovery/VFeatureDiscovery.ts [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+        style: this.contentStyle
+      }), [
+        this.genWrapper(),
+        this.genActions()
+      ])
+    },
+    genActions (): VNode {
+      return this.$createElement('div', {
+        staticClass: 'v-feature-discovery__actions'
+      }, this.$scopedSlots.actions ? this.$scopedSlots.actions({
```
Comment (KaelWD) [inline packages/vuetify/src/components/VFeatureDiscovery/VFeatureDiscovery.ts]: There's a util function for this now

File: packages/vuetify/src/components/VFeatureDiscovery/VFeatureDiscovery.ts [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+        }
+      }) : this.$slots.actions)
+    },
+    genWrapper (): VNode {
+      return this.$createElement('div', {
+        staticClass: 'v-feature-discovery__wrapper'
+      }, [
+        this.$createElement('div', {
+          staticClass: 'v-feature-discovery__title'
+        }, this.$slots.title),
```
Comment (KaelWD) [inline packages/vuetify/src/components/VFeatureDiscovery/VFeatureDiscovery.ts]: Don't use $slots

File: packages/vuetify/src/components/VFeatureDiscovery/VFeatureDiscovery.ts (hunk 1 above)
Comment (sh7dm) [reply packages/vuetify/src/components/VFeatureDiscovery/VFeatureDiscovery.ts]: изображение
Translation: Property style doesn't exist on type Element

Comment (KaelWD) [reply packages/vuetify/src/components/VFeatureDiscovery/VFeatureDiscovery.ts]: Use HTMLElement

**Comment by DRoet** [comment]
FYI the pulsing animation seems a bit broken on IE11, was there more planned for this feature? (since the PR has been stale for a while)

**Comment by mimischi** [comment]
I noticed that the visuals are quite different to those of Materialize. In the latter the highlight is put around the original element, whereas in the current version of this PR the clicked element seems to appear a second time (best visible when dismissing the overlay).

Materialize and Vuetify comparison

**Comment by johnleider** [comment]
This did not make the cut for v2.0 but will come in a minor release.

**Comment by peluprvi** [comment]
Please check https://github.com/vuetifyjs/vuetify/pull/7467 for a new approach on this.

**Comment by peluprvi** [comment]
Closed in favor of https://github.com/vuetifyjs/vuetify/pull/7467

**Pull request closed without merging**